rex/multi_year_resource.py

```diff
@@ -123,15 +127,54 @@ def time_index(self):
         pandas.DatatimeIndex
         """
         if self._time_index is None:
-            for year in self.years:
-                h5 = self[year]
+            for h5 in self._h5_map['h5'].unique():
                 if self._time_index is None:
                     self._time_index = h5.time_index
                 else:
-                    self._time_index = self._time_index.append(h5.time_index)
+                    ti = self._time_index.append(h5.time_index)
+                    self._time_index = ti
 
         return self._time_index
 
+    @staticmethod
+    def _map_file_instances(file_paths, years, res_cls=Resource,
+                            **res_cls_kwargs):
+        """
+        Open all .h5 files and map the open h5py instances to the
+        associated file paths
+
+        Parameters
+        ----------
+        file_paths : list
+            List of filepaths for this handler to handle.
+
+        Returns
+        -------
+        h5_map : pd.DataFrame
+            DataFrame mapping file paths to open resource instances and
+            datasets per file (columns: fp, h5, and dsets)
+        """
+
+        h5_map = pd.DataFrame({'fp': file_paths, 'year': years, 'h5': None})
+        h5_map = h5_map.sort_values('year').reset_index(drop=True)
+
+        if len(h5_map['year'].unique()) < len(h5_map):
+            del res_cls_kwargs['hsds']  # no multi file res on hsds
```

Comment:
Will there never be multi-file res on hsds? Is it fundamentally disallowed or is this just a reflection of the current state?

Reply:
loading data into HSDS is basically an exercise in indexing (as far as i can tell), so all files for a given domain get bundled into a single HSDS "filepath".